File: my-app/src/app/common/season-dto.ts
import {MatchDto} from "./match-dto";

export class SeasonDto {
  public id =0;
  public deleted: boolean = false;
  public currentSeason: boolean = false;
    constructor(public name: string,public start_date:Date,public end_date:Date,public headToHead:number,public numberOfStandings:number, public leagueId: number) {
  }
}

File: FootballApp/src/main/java/com/football/dev/footballapp/dto/SeasonDto.java
package com.football.dev.footballapp.dto;

import com.football.dev.footballapp.models.League;
import com.football.dev.footballapp.models.Match;
import com.football.dev.footballapp.models.Round;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class SeasonDto {
    private Long id;
    private String name;
    private boolean currentSeason;
    private LocalDateTime start_date;
    private LocalDateTime end_date;
    private Integer headToHead;
    private Integer numberOfStandings;
//    private LeagueDTO league;

    public SeasonDto(Long id, String name, boolean currentSeason) {
        this.id = id;
        this.name = name;
        this.currentSeason = currentSeason;
//        this.league = league;
    }
    public SeasonDto(String name) {
        this.name = name;

    }
}
